core/merginapi.cpp

```diff
@@ -1693,7 +1733,15 @@ void MerginApi::finalizeProjectPullCopy( const QString &projectFullName, const Q
       CoreUtils::log( "pull " + projectFullName, "Failed to open temp file for reading " + item.tempFileName );
       return;
     }
-    f.write( fTmp.readAll() );
+    QByteArray data = fTmp.readAll();
+    qint64 bytesWritten = f.write( data );
+
+    // not enough space on file system
+    if ( data.size() != bytesWritten )
+    {
+      CoreUtils::log( "pull " + projectFullName, "Failed to write content into: " + fTmp.fileName() );
+      return;
```

Comment:
when we return here
1/ file `f` is not closed
2/ file `f` is still present in the final folder in some random bad shape

When there is problem with sync (any), we need to guarantee that 
1/ all files are in the original/pre-update/previous version 
2/ there are no file leftovers anywhere, both in temporary folders and updated folders. 
3/ we have note in diagnostics logs about the issues